llvm/lib/Transforms/Vectorize/VPRecipeBuilder.h

```diff
@@ -149,6 +149,16 @@ class VPRecipeBuilder {
                                           : std::make_optional(It->second);
   }
 
+  bool setScalingForReduction(const Instruction *ExitInst,
+                              unsigned ScaleFactor) {
+    auto It = ScaledReductionMap.find(ExitInst);
+    if (It != ScaledReductionMap.end()) {
```

Comment:
Is it better to just assert that there is an entry instead? I think that it would be a mistake to call `setScalingForReduction` for an instruction that didn't correspond to scaled reduction.